pkg/controllers/core/catalog_controller.go

```diff
@@ -358,3 +366,92 @@ func (r *CatalogReconciler) syncPackages(ctx context.Context, declCfg *declcfg.D
 	}
 	return nil
 }
+
+// syncCatalogMetadata will create a `CatalogMetadata` resource for each
+// "olm.bundle" and "olm.package" objects that exist for the given catalog contents. Returns an
+// error if any are encountered.
+func (r *CatalogReconciler) syncCatalogMetadata(ctx context.Context, fs fs.FS, catalog *v1alpha1.Catalog) error {
```

Comment:
Nit: avoid shadowing the `fs` package name with an `fs` variable
```suggestion
func (r *CatalogReconciler) syncCatalogMetadata(ctx context.Context, fsys fs.FS, catalog *v1alpha1.Catalog) error {
```